data_steward/validation/participants/readers.py

```diff
@@ -0,0 +1,273 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+A module to perform reading operations for validation data.
+"""
+# Python imports
+import logging
+
+# Third party imports
+
+# Project imports
+import bq_utils
+import resources as rc
+import constants.validation.participants.readers as consts
+
+LOGGER = logging.getLogger(__name__)
+
+
+def _get_utf8_string(value):
+    result = None
+    try:
+        result = value.encode('utf-8', 'ignore')
+    except AttributeError:
+        if value is None:
+            LOGGER.debug("Value '%s' can not be utf-8 encoded", value)
+        elif isinstance(value, int):
+            result = str(value)
+
+    return result
+
+
+def create_match_values_table(project, rdr_dataset, destination_dataset):
+    """
+    Get the desired matching values from the combined observation table.
+
+    This retrieves all possible matching values from the observation table
+    used in participant matching.  Returned query data is limited to the
+    person_id, observation_concept_id, and the value_as_string fields.
+
+    :param project:  The project name to query
+    :param rdr_dataset:  The rdr dataset name to query
+    :param destination_dataset:  The dataset to write the result table to
+
+    :return: The name of the interim table
+    """
+    query_string = consts.ALL_PPI_OBSERVATION_VALUES.format(
+        project=project,
+        dataset=rdr_dataset,
+        table=consts.OBSERVATION_TABLE,
+        pii_list=', '.join(consts.PII_CODES_LIST)
+    )
+
+    LOGGER.debug("Participant validation ran the query\n%s", query_string)
+    results = bq_utils.query(
+        query_string,
+        destination_dataset_id=destination_dataset,
+        destination_table_id=consts.ID_MATCH_TABLE,
+        write_disposition='WRITE_TRUNCATE',
+        batch=True
+    )
+
+    query_job_id = results['jobReference']['jobId']
+    incomplete_jobs = bq_utils.wait_on_jobs([query_job_id])
+    if incomplete_jobs != []:
+        raise bq_utils.BigQueryJobWaitError(incomplete_jobs)
+
+    return consts.ID_MATCH_TABLE
+
+
+def get_ehr_person_values(project, dataset, table_name, column_name):
+    """
+    Get the desired matching values from the combined observation table.
+
+    This retrieves all possible matching values from the observation table
+    used in participant matching.  Returned query data is limited to the
+    person_id, observation_concept_id, and the value_as_string fields.
+
+    :param project: The name of the project to query for ehr values
+    :param dataset:  The name of the dataset to query for ehr values
+    :param table_name:  The name of the table to query for ehr values
+    :param concept_id: the id of the concept to identify with ehr data
+    :param id_list:  list of person_id values to limit the result set to
+
```

Comment:
unknown parameters in doc string.

Reply:
concept_id and id_list removed.  column_name added.